crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -481,8 +483,16 @@ impl Object {
     }
 
     /// Attempts to convert the object into a MovePackage
-    async fn as_move_package(&self) -> Option<MovePackage> {
-        MovePackage::try_from(self).ok()
+    async fn as_move_package(&self, ctx: &Context<'_>) -> Option<MovePackage> {
+        let Some(checkpoint_viewed_at) = match self.checkpoint_viewed_at {
```

Comment:
Potentially this is temporary and eventually `Object` will also get a non-optional `self.checkpoint_viewed_at`, but it's a little strange that we need to do a fetch of the latest checkpoint sequence number here.